### DISCUSSION TITLE
Help a user to choose from a long list

### DISCUSSION THREAD

**Discussion by willrichards** [UKHomeOffice/design-system#401] [Ideas]
I've observed different teams researching the best way for a user to make the right selection from a long list, often resulting in repeat findings.

My suggestion is to compile some guidance on 'Help users to choose from a long list' to reduce the duplication, and to help teams select the right method for their scenario first time.

This might include guidance around:
- when, and when not to use patterns
- research findings on what makes the patterns work well
- content advice on what to do with long lists of data, and how to optimise for selection

I have the following questions:

1. Would guidance on this topic have value for teams?
2. Are teams willing to contribute examples to this discussion on GitHub, showing what works well and what doesn't?
3. Any other feedback or suggestions welcome

**Comment by robertdeniszczyc2** [comment]
Hi Will,

Guidance around this would definitely have value. I wonder if one consideration around this is the context which the list is in, and what defines a "long list"? Is 10 items classed as long, or 10,000?

If there are thousands of items in the list, is a list the right pattern to use for displaying the information?

Thanks,
Rob

**Comment by tomyems** [comment]
I think we should cover what we mean by long in the guidance. Perhaps with less than 10, about 100 and 1,000? There isn't going to be one answer, but guidance about the advantages and disadvantages of different designs would be helpful for designers.

**Comment by guermanc** [comment]
Some interesting observations on accessibility of listboxes from [ARIA Authoring Practices Guide Listbox guidance](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/), which will be especially relevant to long lists of options:

- Avoid long option names
- Don't use option names that start with the same word or phrase

There's the optgroup element for dividing up groups of options but screen reader support is not great https://a11ysupport.io/tech/html/optgroup_element

**Comment by GenevieveParke** [comment]
In UKVI we use a 'type ahead' pattern when we want users to select their country of nationality
![type ahead](https://user-images.githubusercontent.com/55453531/179504785-36588ca5-46c8-4c2a-9126-8c6eb03f0826.png)

**Comment by daniel-ac-martin** [comment]
In case it is relevant: https://github.com/alphagov/accessible-autocomplete
(Might be a little neglected, but maybe there is some good design stuff in there.)

**Comment by larab-codes** [comment]
![image](https://user-images.githubusercontent.com/68660397/214084220-535f3954-d1f4-4cdf-a649-9227c0a5c253.png)
![image](https://user-images.githubusercontent.com/68660397/214084260-d81923d6-6a13-45df-9f62-74d1e95ca06b.png)
![image](https://user-images.githubusercontent.com/68660397/214084317-e09e6b90-b0ef-4032-afdd-4a951b8f64c6.png)

**Comment by LauraLian** [comment]
[LinkedIn blog post with a design using a long list of radio buttons ](https://www.linkedin.com/posts/adamsilverhq_a-few-days-ago-i-shared-a-design-containing-activity-7197584090278019072-qCs1/?utm_source=share&utm_medium=member_desktop)(with hint text) instead of dropdowns or typeaheads from Adam Silver.

**Reply by LauraLian** [reply]
![image](https://github.com/UKHomeOffice/design-system/assets/159554120/04febc88-bd0d-4333-a638-e4369b6f165a)

**Reply by LauraLian** [reply]
Content from the email from Adam Silver (not my content):

> Last week I posted this on LinkedIn:

===

I tell my students to avoid select boxes.

Because it’s often better to go for radios.

But they often say “But it’ll make the list long”.

Yep, but that doesn’t necessarily mean it’s bad UX.

See below for the page I designed to let users select a course. Huge list of radio buttons. But no issues in user research whatsoever.
​
Does this mean you should always use radio buttons?
​
No.
​
But most designers would balk at a design like this (even though this design worked well for my users).

Long list of radio buttons for users to select a course
===

Many designers and researchers criticised my design in the comments. Some even accused me of breaking a UX law.

So today I’ll respond to the 6 main criticisms, to give you all the ammo you need to try this out yourself.

Here goes:

“How did you test it?”
We asked users to make an offer.

To make an offer you have to go through a multi-step flow. One step involves selecting a course from the long list of radio buttons.

The result:

Everyone chose a course quick and easily.

But what people really meant was:

“What other options did you test? Just because this worked, doesn’t mean it’s best!”
I rarely test 2 solutions at the same time.

Don’t get me wrong: I consider many options.

But only one gets tested (unless it fails, then I try another).

That’s because there’s usually a clear winner. Plus testing 2 versions has downsides.

But even if there was a better design out there, given how well the radio buttons worked, it would be a lot of effort to shave just a few milliseconds off the interaction.

Plus the product had many other more important issues to solve.

“I would use a dropdown.”
Most designers love a dropdown (select box):


That’s because it’s the easy choice to keep your UI clean and your page short.

But select boxes have downsides:

Some users try to type into them
Some users confuse focused items for selected items
You cannot pinch-zoom on mobile
You cannot add hint text (like I did with the radio buttons)
But even if you ignore these issues, select boxes don’t make the list shorter. They just hide it behind an extra click.

“What about an autocomplete?”
In case you don’t know, an autocomplete control is a text box that reveals options as the user types.


But autocompletes have downsides:

Some users may not know exactly what to type
Some users may not realise they can reveal all the options
Some users won’t spot the suggestions as they type (they look at the keyboard)
All of which degrades UX and increases the time it takes to select an option.

Don’t get me wrong, I would have explored this pattern if radio buttons performed badly.

But they didn’t.

“There’s too much scrolling, especially on mobile”
We didn’t test on mobile because our research showed that users were all on desktop.

But that’s a copout so let me address the point directly:

There’s nothing else on screen but radio buttons, so there’s nothing users need to scroll past (the radio buttons are the UI)
Scrolling a web page is no harder than scrolling an overlaid list of options (like you’d get with a select box or autocomplete)
It’s quick and easy to scroll on mobile (one or two flicks max)
“You broke Miller’s Law”
Miller’s law states:

On average people cannot hold more than 7 items in their short term memory.
So because I used 20+ radio buttons some people think I broke the law.

But users don’t need to store them in their short term memory. They’re literally on screen.

So that’s the main criticisms addressed.

But what do they tell you?

Here’s what they tell me:

Boring design is not always popular with designers. But it’s almost always good for users.

If you’d like to learn how to design simple, accessible and rebellious form flows that break the law (and actually work), you might like my course, Form Design Mastery:

​[https://formdesignmastery.com/](https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fclick.convertkit-mail2.com%2Fgku28qn880b5hd5vv7rsrh5qor999cm%2Fkkhmh6hlzo78z8ul%2FaHR0cHM6Ly9mb3JtZGVzaWdubWFzdGVyeS5jb20v&data=05%7C02%7Claura.lian%40digital.homeoffice.gov.uk%7Cfaaba751f6214935c26708dc77d1effb%7C90a6982e16354aa087a6d5d63c7bc66e%7C0%7C0%7C638516991648438450%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C40000%7C%7C%7C&sdata=9%2BG2sQs%2F%2FeeSaHM4BzaKuHU0LdZVHPJV3O44%2Bh%2BJsss%3D&reserved=0)

**Comment by StephanieLongden** [comment]
We have now published the guidance "[Help users to choose from a long list](https://design.homeoffice.gov.uk/patterns/long-lists)" on the Home Office Design System.

If you have done any relevant research or have feedback on the guidance, please share it in this thread, thanks!
